lucille-core/src/main/java/com/kmwllc/lucille/util/FileUtils.java

```diff
@@ -64,11 +71,47 @@ public static boolean isValidURI(String uriString) {
    * @param path the path as a String
    * @return the correct InputStream
    */
-  public static InputStream getInputStream(String path) throws IOException {
+  public static InputStream getInputStream(String path, FileSystemManager fsManager) {
+    InputStream is;
+
+    // setting up VFS
     if (isValidURI(path)) {
-      return VFSInputStream.open(path);
+      FileObject file;
+      try {
+        file = fsManager.resolveFile(path);
+
+        if (file == null) {
+          log.warn("File object is null for path: {}", path);
+          return null;
+        }
+
+        if (!file.exists()) {
```

Comment:
is this existence test necessary, or if the file doesn't exist and we call file.getContent() will we get a FileSystemException that will be caught and logged

Reply:
looking at docs for the individual methods we're calling here, can we simplify the code as follows?

`      try {
        file = fsManager.resolveFile(path); // never returns null
        FileContent content = file.getContent(); // can be called even if !file.exists()
        return content.getInputStream(); // should throw FileSystemException if !file.exists()
      } catch (FileSystemException e) {
        log.warn("Error retrieving file contents: {}", path, e);
        return null;
      }`